src/python/py_generate_pipeline.cpp

```diff
@@ -0,0 +1,359 @@
+            setenv(::get_tokenizers_env_name(), path.c_str(), 1);
+        #endif
+        else
+            was_already_set = true;
+    }
+
+    ~GenAIEnvManager() {
+        if (!was_already_set){
+        #ifdef _WIN32
+            _putenv_s(::get_tokenizers_env_name(), "");
+        #else
+            unsetenv(::get_tokenizers_env_name());
+        #endif
+        }
+    }
+
+private:
+    bool was_already_set;
+};
+
+}
+
+
+namespace py = pybind11;
+using ov::genai::LLMPipeline;
+using ov::genai::Tokenizer;
+using ov::genai::GenerationConfig;
+using ov::genai::EncodedResults;
+using ov::genai::DecodedResults;
+using ov::genai::StopCriteria;
+using ov::genai::StreamerBase;
+using ov::genai::StreamerVariant;
+using ov::genai::OptionalGenerationConfig;
+
+namespace {
+
+void update_config_from_kwargs(GenerationConfig& config, const py::kwargs& kwargs) {
+    if (kwargs.contains("max_new_tokens")) config.max_new_tokens = kwargs["max_new_tokens"].cast<size_t>();
+    if (kwargs.contains("max_length")) config.max_length = kwargs["max_length"].cast<size_t>();
+    if (kwargs.contains("ignore_eos")) config.ignore_eos = kwargs["ignore_eos"].cast<bool>();
+    if (kwargs.contains("num_beam_groups")) config.num_beam_groups = kwargs["num_beam_groups"].cast<size_t>();
+    if (kwargs.contains("num_beams")) config.num_beams = kwargs["num_beams"].cast<size_t>();
+    if (kwargs.contains("diversity_penalty")) config.diversity_penalty = kwargs["diversity_penalty"].cast<float>();
+    if (kwargs.contains("length_penalty")) config.length_penalty = kwargs["length_penalty"].cast<float>();
+    if (kwargs.contains("num_return_sequences")) config.num_return_sequences = kwargs["num_return_sequences"].cast<size_t>();
+    if (kwargs.contains("no_repeat_ngram_size")) config.no_repeat_ngram_size = kwargs["no_repeat_ngram_size"].cast<size_t>();
+    if (kwargs.contains("stop_criteria")) config.stop_criteria = kwargs["stop_criteria"].cast<StopCriteria>();
+    if (kwargs.contains("temperature")) config.temperature = kwargs["temperature"].cast<float>();
+    if (kwargs.contains("top_p")) config.top_p = kwargs["top_p"].cast<float>();
+    if (kwargs.contains("top_k")) config.top_k = kwargs["top_k"].cast<size_t>();
+    if (kwargs.contains("do_sample")) config.do_sample = kwargs["do_sample"].cast<bool>();
+    if (kwargs.contains("repetition_penalty")) config.repetition_penalty = kwargs["repetition_penalty"].cast<float>();
+    if (kwargs.contains("pad_token_id")) config.pad_token_id = kwargs["pad_token_id"].cast<int64_t>();
+    if (kwargs.contains("bos_token_id")) config.bos_token_id = kwargs["bos_token_id"].cast<int64_t>();
+    if (kwargs.contains("eos_token_id")) config.eos_token_id = kwargs["eos_token_id"].cast<int64_t>();
+    if (kwargs.contains("eos_token")) config.eos_token = kwargs["eos_token"].cast<std::string>();
+    if (kwargs.contains("bos_token")) config.bos_token = kwargs["bos_token"].cast<std::string>();
```

Comment:
can we define whole `generation _config` via `kwargs` ?

Reply:
If you mean `pipe.generate(input_ids, generation_config=config)` then at the moment it's not possible but it's easy to add. 
Let me add such thing as well